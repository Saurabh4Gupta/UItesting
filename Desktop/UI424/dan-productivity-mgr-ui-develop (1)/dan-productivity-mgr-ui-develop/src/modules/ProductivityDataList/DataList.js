/* eslint-disable no-shadow */
/* eslint-disable react/prop-types */
/* eslint-disable no-unused-vars */
import React, { useState, useEffect, useRef, useContext } from 'react';
import { Box, Tabs } from '@dentsu-ui/components';
import PropTypes from 'prop-types';
import Toast from '@dentsu-ui/components/dist/cjs/components/Toast';
import { useMutation } from '@apollo/client';
import EmptyTable from './EmptyTable';
import TableList from './TableList';
import Loader from '../../components/loading';
import { dataFieldCms as PageContent } from '../../cms';
import { DELETE_DATA_REQUEST } from '../CreateData/mutation';
import { EDIT_DATA_REQUEST_STATUS } from './mutation';
import { FlagIt } from '../../contexts/marketOptions';

const DataList = (props) => {
  const {
    cmsData,
    clientCode,
    market,
    dataList,
    setDataList,
    loading,
    completeDataList,
    setCompleteDataList,
    handleModal,
    refetch,
    refetchCompletedata,
    // setMarket,
  } = props;
  const [deleteModalData, setIsDeleteModal] = useState({
    isDeleteModal: false,
    requestId: undefined,
  });
  const [selectedFilter, setSelectedFilter] = useState([
    { key: 'totalTenure', value: '' },
  ]);
  const [searchInput, setSearchInput] = useState('');
  const [tabIndex, setTabIndex] = useState(0);
  const { dataList: data, totalCount } = dataList;
  const { dataList: completeData, totalCount: completeCount } = completeDataList

  const [moveToCompleteModelData, setIsMoveToCompleteModel] = useState({
    isMoveToComplete: false,
    requestID: undefined,
  });
  const [deleteDataRequest, { data: deleteData }] = useMutation(
    DELETE_DATA_REQUEST,
  );
  const [editDataRequestStatus, { data: dataStatus }] = useMutation(
    EDIT_DATA_REQUEST_STATUS,
  )

  const toast = Toast();

  // const initialRender = useRef(true);
  useEffect(() => {
    if (deleteData) {
      const { status } = deleteData.deleteDataRequests;
      if (status !== 200) {
        return toast({
          title: PageContent.failurNotificationMsg,
          status: 'error',
        });
      }
      refetch();
      refetchCompletedata();
      return toast({
        title: PageContent.toastRequestDeleted,
        status: 'success',
      });
    }
  }, [deleteData]);

  useEffect(() => {
    if (dataStatus) {
      const { status } = dataStatus.editDataRequestStatus;
      if (status !== 200) {
        return toast({
          title: PageContent.failurNotificationMsg,
          status: 'error',
        });
      }
      refetch();
      refetchCompletedata();
    }
  }, [dataStatus]);

  // useEffect(() => {
  //   if (initialRender.current) {
  //     initialRender.current = false;
  //   } else if (tabIndex === 0) {
  //     setDataList(getData(market.value, 'ongoing'));
  //     setCompleteDataList(getData(market.value, 'complete'));
  //   } else {
  //     setCompleteDataList(getData(market.value, 'complete'));
  //     setDataList(getData(market.value, 'ongoing'));
  //   }
  // }, [market, tabIndex]);

  const getFilteredList = (data, searchInput, filterInput) => {
    let updatedList = data.filter(
      (d) => d.clientMarket.toLowerCase().includes(searchInput.toLowerCase())
        || d.name.toLowerCase().includes(searchInput.toLowerCase()),
    );
    if (filterInput !== '') {
      updatedList = updatedList.filter((key) => key.totalTenure.includes(filterInput));
    }
    return updatedList;
  };

  // useEffect(() => {
  //   const requestStatus = tabIndex === 0 ? 'ongoing' : 'complete';
  //   const allDataList = getData(market.value, requestStatus);
  //   const filteredList = getFilteredList(
  //     allDataList.data,
  //     searchInput,
  //     selectedFilter[0].value,
  //   );
  //   if (tabIndex === 0) {
  //     setDataList({ data: filteredList, totalCount: allDataList.totalCount });
  //   }
  //   if (tabIndex === 1) {
  //     setCompleteDataList({
  //       data: filteredList,
  //       totalCount: allDataList.totalCount,
  //     });
  //   }
  // }, [searchInput, selectedFilter]);
  const userPermissionInfo = useContext(FlagIt);

  let userCreatePermission = false;
  userPermissionInfo.forEach(item => {
    if (item.resource === 'PDR') {
      userCreatePermission = true;
    }
  })
  const searchChangeHandler = (input) => {
    setSearchInput(input);
  };

  const handleFilter = (input) => {
    setSelectedFilter(input);
  };

  const handleDelete = (requestId) => {
    deleteDataRequest({ variables: { id: requestId } });
  };

  const handleMoveToCompleteData = (requestId) => {
    editDataRequestStatus({ variables: { data: { id: requestId, isComplete: true } } })

    return toast({
      title: '',
      content: PageContent.toastMovedToComplete,
      status: 'success',
    });
  };

  const handleMoveToOngoing = (requestId) => {
    editDataRequestStatus({ variables: { data: { id: requestId, isComplete: false } } })

    return toast({
      title: '',
      content: PageContent.toastMovedToOngoing,
      status: 'success',
    });
  };

  const handleTabIndex = (index) => {
    setTabIndex(index);
  };
  const handleDeleteModel = (value) => {
    setIsDeleteModal({ isDeleteModal: true, requestId: value });
  };
  const handleMoveToCompleteModel = (value) => {
    setIsMoveToCompleteModel({ isMoveToComplete: true, requestID: value });
  };
  return (
    <>
      <Box mt="30px">
        <br />
        <Tabs>
          <Tabs.List>
            <Tabs.Tab label={cmsData.ongoingLabel} count={totalCount} />
            <Tabs.Tab
              label={cmsData.completeLabel}
              count={completeCount}
            />
          </Tabs.List>
          {loading ? (
            <Loader />
          ) : (
            <Tabs.Panels>
              <Tabs.Panel>
                {dataList.totalCount > 0 ? (
                  <TableList
                    data={data}
                    cmsData={cmsData}
                    deleteModalData={deleteModalData}
                    setIsDeleteModal={setIsDeleteModal}
                    setIsMoveToCompleteModel={setIsMoveToCompleteModel}
                    moveToCompleteModelData={moveToCompleteModelData}
                    actionName={
                        tabIndex === 0
                          ? cmsData.moveToComplete
                          : cmsData.moveToOnGoing
                      }
                    handleDelete={handleDelete}
                    clientCode={clientCode}
                    search={searchChangeHandler}
                    handleDeleteModel={handleDeleteModel}
                    handleMoveToCompleteModel={handleMoveToCompleteModel}
                    handleMoveToCompleteData={handleMoveToCompleteData}
                    showStatus={tabIndex === 0}
                    handleFilter={handleFilter}
                    selectedFilter={selectedFilter}
                  />
                  ) : (
                    userCreatePermission === true && (
                      <EmptyTable
                        defaultText={cmsData.emptyProductivityDatarequestCaption}
                        handleModal={handleModal}
                      />
                    )
                    )}
              </Tabs.Panel>
              <Tabs.Panel>
                {completeDataList.totalCount > 0 ? (
                  <TableList
                    data={completeData}
                    cmsData={cmsData}
                    actionName={cmsData.moveToOnGoing}
                    handleMoveToOngoing={handleMoveToOngoing}
                    showStatus={false}
                    clientCode={clientCode}
                    search={searchChangeHandler}
                    handleFilter={handleFilter}
                    selectedFilter={selectedFilter}
                  />
                  ) : (
                    userCreatePermission === true && (
                    <EmptyTable
                      defaultText={
                          cmsData.emptyCompletedProductivityDatarequestCaption
                        }
                      handleModal={handleModal}
                    />
                    )
                    )}
              </Tabs.Panel>
            </Tabs.Panels>
            )}
        </Tabs>
      </Box>
    </>
  );
};

DataList.propTypes = {
  cmsData: PropTypes.object,
  clientCode: PropTypes.string,
  market: PropTypes.object || PropTypes.string,
  dataList: PropTypes.object,
  setDataList: PropTypes.func,
  loading: PropTypes.bool,
};
DataList.defaultProps = {
  cmsData: {},
  market: { value: '' },
  dataList: {},
  clientCode: '',
  setDataList: () => { },
  loading: true,
};
export default DataList;
