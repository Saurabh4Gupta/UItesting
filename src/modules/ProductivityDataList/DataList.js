/* eslint-disable no-unused-vars */
import React from 'react';
import { Box, Tabs } from '@dentsu-ui/components';
import PropTypes from 'prop-types';
import EmptyTable from './EmptyTable';
import TableList from './TableList';

const DataList = (props) => {
  const {
    cmsData,
    handleModal,
    ongoingDataList,
    completeDataList,
    handleToggleData,
    tabIndex,
    handleTabIndex,
    handleDeleteModel,
  } = props;
  const { data, totalCount } = ongoingDataList;
  const { completedData, completedCount } = completeDataList;

  return (
    <Box mt="30px">
      <Tabs onChange={handleTabIndex} index={tabIndex}>
        <Tabs.List>
          <Tabs.Tab label={cmsData.ongoingLabel} count={data.length} />
          <Tabs.Tab
            label={cmsData.completeLabel}
            count={completedData.length}
          />
        </Tabs.List>
        <Tabs.Panels>
          <Tabs.Panel>
            {data.length > 0 ? (
              <TableList
                data={data}
                cmsData={cmsData}
                handleToggleData={handleToggleData}
                actionName={cmsData.moveToComplete}
                handleDeleteModel={handleDeleteModel}
              />
            ) : (
              <EmptyTable
                defaultText={cmsData.emptyProductivityDatarequestCaption}
                handleModal={handleModal}
              />
            )}
          </Tabs.Panel>
          <Tabs.Panel>
            {completedData.length > 0 ? (
              <TableList
                data={completedData}
                cmsData={cmsData}
                actionName={cmsData.moveToOnGoing}
                handleToggleData={handleToggleData}
                showStatus={false}
              />
            ) : (
              <EmptyTable
                defaultText={
                  cmsData.emptyCompletedProductivityDatarequestCaption
                }
                handleModal={handleModal}
              />
            )}
          </Tabs.Panel>
        </Tabs.Panels>
      </Tabs>
    </Box>
  );
};

DataList.propTypes = {
  cmsData: PropTypes.object,
  handleModal: PropTypes.func,
  ongoingDataList: PropTypes.object,
  completeDataList: PropTypes.object,
  handleToggleData: PropTypes.func.isRequired,
  tabIndex: PropTypes.number.isRequired,
  handleTabIndex: PropTypes.func.isRequired,
  handleDeleteModel: PropTypes.func.isRequired,
};
DataList.defaultProps = {
  cmsData: {},
  handleModal: () => {},
  ongoingDataList: {},
  completeDataList: [],
};
export default DataList;
