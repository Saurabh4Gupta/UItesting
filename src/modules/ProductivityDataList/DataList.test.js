import React from 'react';
import { shallow } from 'enzyme';
import DataList from './DataList';

describe('<EmptyState />', () => {
  let wrapper;
  let wrapper2;
  let wrapper3;

  const props = {
    setCompleteDataList: jest.fn(),
    tabIndex: 0,
    mockData: {
      data: [
        {
          id: 1,
          client: 'Microsoft',
          localMarket: { value: 'UK', label: 'United Kingdom' },
          status: 'Overdue',
          clientMarket: 'Microsoft United Kingdom',
          name: 'Productivity Q2 2020',
          briefing: 'Lorem Ipsum.',
          dueDate: '13/07/2020',
          updatedAt: '29/10/20 at 14:32',
          createdAt: '2020-08-14',
          assignTo: [
            {
              value: 'ryanKillick',
              label: 'Ryan Killick',
              userImage: 'user1.png',
            },
          ],
          reportingYear: {
            value: 'April 2020  -  March 2021',
            label: 'April 2020  -  March 2021',
          },
          actualData: { value: 4, label: '4 months' },
          forecastData: { value: 8, label: '8 months' },
          isActive: true,
          isDeleted: false,
          year: '2020',
          quarter: 'Q2',
          totalTenure: '2020 Q2',
          isCompleted: false,
        },
      ],
    },
    cmsData: 'data',
    handleModal: jest.fn(),
    handleToggleData: jest.fn(),
    handleTabIndex: jest.fn(),
    loading: false,
    clientCode: 'MC',
    market: {
      value: '',
      label: 'All markets',
    },
    completeDataList: {
      totalCount: 1,
      data: [],
    },
    dataList: {
      totalCount: 1,
      data: {
        id: 1,
        client: 'Microsoft',
        localMarket: { value: 'UK', label: 'United Kingdom' },
        status: 'Overdue',
        clientMarket: 'Microsoft United Kingdom',
        name: 'Productivity Q2 2020',
        briefing: 'Lorem Ipsum',
        dueDate: '13/07/2020',
        updatedAt: '29/10/20 at 14:32',
        createdAt: '2020-08-14',
        assignTo: [],
        reportingYear: {
          value: 'April 2020  -  March 2021',
          label: 'April 2020  -  March 2021',
        },
        actualData: { value: 4, label: '4 months' },
        forecastData: { value: 8, label: '8 months' },
        isActive: true,
        isDeleted: false,
        year: '2020',
        quarter: 'Q2',
        totalTenure: '2020 Q2',
      },
      setDataList: jest.fn(),
      setCompleteDataList: jest.fn(),
      setMarket: jest.fn(),
    },
  };
  const prop2 = {
    cmsData: 'data',
    handleModal: jest.fn(),
    handleToggleData: jest.fn(),
    tabIndex: 0,
    isCompleted: false,
    loading: true,
    handleTabIndex: jest.fn(),
    handleDeleteModel: jest.fn(),
    clientCode: 'MC',
    market: {
      value: '',
      label: 'All markets',
    },
    completeDataList: {
      totalCount: 1,
      data: ['data'],
    },
    dataList: {
      totalCount: 1,
      data: {
        id: 1,
        client: 'Microsoft',
        localMarket: { value: 'UK', label: 'United Kingdom' },
        status: 'Overdue',
        clientMarket: 'Microsoft United Kingdom',
        name: 'Productivity Q2 2020',
        briefing: 'Lorem Ipsum',
        dueDate: '13/07/2020',
        updatedAt: '29/10/20 at 14:32',
        createdAt: '2020-08-14',
        assignTo: [],
        reportingYear: {
          value: 'April 2020  -  March 2021',
          label: 'April 2020  -  March 2021',
        },
        actualData: { value: 4, label: '4 months' },
        forecastData: { value: 8, label: '8 months' },
        isActive: false,
        isDeleted: false,
        year: '2020',
        quarter: 'Q2',
        totalTenure: '2020 Q2',
        isCompleted: false,
      },
      setDataList: jest.fn(),
      setCompleteDataList: jest.fn(),
      setMarket: jest.fn(),
    },
  };
  const prop3 = {
    cmsData: 'data',
    handleModal: jest.fn(),
    handleToggleData: jest.fn(),
    tabIndex: 0,
    isCompleted: false,
    loading: false,
    handleTabIndex: jest.fn(),
    handleDeleteModel: jest.fn(),
    clientCode: 'MC',
    market: {
      value: '',
      label: 'All markets',
    },
    completeDataList: {
      totalCount: 0,
      data: [],
    },
    dataList: {
      totalCount: 0,
      data: [],
    },
    setDataList: jest.fn(),
    setCompleteDataList: jest.fn(),
    setMarket: jest.fn(),
  };
  beforeAll(() => {
    wrapper = shallow(<DataList {...props} />);
    wrapper2 = shallow(<DataList {...prop2} />);
    wrapper3 = shallow(<DataList {...prop3} />);
  });
  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
    expect(wrapper2).toMatchSnapshot();
    expect(wrapper3).toMatchSnapshot();
  });
  it('handleModal to be defined', () => {
    const handler = wrapper.find('CreateData').prop('handleModal');
    expect(handler()).not.toBeNull();
  });
  it('handleTabIndex to be defined', () => {
    const handler = wrapper.find('Tabs').prop('onChange');
    expect(handler()).not.toBeNull();
  });
  it('handleDelete to be defined', () => {
    const handler = wrapper.find('TableList').at(0).prop('handleDelete')(1);
    expect(handler).not.toBeNull();
  });
  it('handleDeleteModel to be defined', () => {
    const handler = wrapper.find('TableList').at(0).prop('handleDeleteModel');
    expect(handler()).not.toBeNull();
  });
  it('handleMoveToCompleteModel to be defined', () => {
    const handler = wrapper
      .find('TableList')
      .at(0)
      .prop('handleMoveToCompleteModel');
    expect(handler()).not.toBeNull();
  });
  it('searchChangeHandler to be defined', () => {
    const handler = wrapper.find('TableList').at(1).prop('search');
    expect(handler()).not.toBeNull();
  });
  it('addRequest to be defined', async () => {
    const handler = await wrapper.find('CreateData').prop('addRequest')({
      localMarket: { label: 'test' },
    });
    expect(handler).toBeNull();
  });
  it('handleMoveToOngoing to be defined', () => {
    const handler = wrapper.find('TableList').at(1).prop('handleMoveToOngoing')(
      1,
    );
    expect(handler).not.toBeNull();
  });
  it('handleMoveToCompleteData', () => {
    const handler = wrapper
      .find('TableList')
      .at(0)
      .prop('handleMoveToCompleteData')(1);
    expect(handler).not.toBeNull();
  });
});
