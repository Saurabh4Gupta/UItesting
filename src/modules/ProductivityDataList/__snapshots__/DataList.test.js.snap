// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EmptyState /> should match snapshot 1`] = `
<Fragment>
  <CreateData
    addRequest={[Function]}
    cmsData="data"
    handleModal={[Function]}
    isModalOpen={false}
    market={
      Object {
        "value": "",
      }
    }
  />
  <Box
    mt="30px"
  >
    <Tabs
      index={0}
      onChange={[Function]}
    >
      <ForwardRef>
        <ForwardRef
          count={0}
        />
        <ForwardRef
          count={0}
        />
      </ForwardRef>
      <Loader />
    </Tabs>
  </Box>
</Fragment>
`;
